cmd/celvet/main.go

```diff
@@ -81,13 +84,43 @@ func main() {
 	}
 
 	limitErrors := celvet.CheckMaxLimits(structural)
+	costErrors, compileErrors := celvet.CheckExprCost(structural)
+	if *useJSON {
+		emitJSON(limitErrors, costErrors, compileErrors)
+	} else {
+		emitNormal(limitErrors, costErrors, compileErrors, *humanReadable)
+	}
+
+	if len(limitErrors)+len(costErrors)+len(compileErrors) > 0 {
+		os.Exit(1)
+	}
+}
+
+func emitJSON(limitErrors []error, costErrors []*celvet.CostError, compileErrors []error) {
+	type JSONOutput struct {
+		LimitErrors   []error             `json:"limitErrors"`
+		CostErrors    []*celvet.CostError `json:"costErrors"`
+		CompileErrors []error             `json:"compileErrors"`
+	}
+
+	buf := bytes.NewBuffer(nil)
+	encoder := json.NewEncoder(buf)
+	encoder.SetEscapeHTML(false)
+	output := JSONOutput{LimitErrors: limitErrors, CostErrors: costErrors, CompileErrors: compileErrors}
+	err := encoder.Encode(output)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error generating JSON output: %s\n", err)
+	}
+	// use Printf instead of Println to prevent a redundant newline from being output
+	fmt.Printf("%s", buf.Bytes())
```

Comment:
By redundant to you mean there were two newlines being printed when Println was used?

Reply:
Right - looking at the contents of `buf`, the last character is `10`\`\n`, which makes me think that the `json` module inserts one itself when encoding.